config/logs.js

```diff
@@ -0,0 +1,114 @@
+const _ = require('lodash');
+
+module.exports = {
+  hidFormatter(level, msg, logObject) {
+    // Clone the logged object to avoid mutating important objects (e.g. request)
+    // in case the code continues executing after being logged.
+    const localLogObject = _.clone(logObject || {});
+
+    // Define our meta with some defaults.
+    const metadata = {};
+    metadata.level = level;
+
+    // Log IP if we have it.
+    if (logObject.request && logObject.request.info && logObject.request.info.remoteAddress) {
+      metadata.ip = localLogObject.request.info.remoteAddress;
+    }
+    if (logObject.request && logObject.request.headers && logObject.request.headers['x-forwarded-for']) {
+      metadata.ip = localLogObject.request.headers['x-forwarded-for'];
+    }
+
+    // Include custom user object from log
+    metadata.user = localLogObject.user || {};
+
+    // Include custom oauth object from log
+    metadata.oauth = localLogObject.oauth || {};
+
+    // If this is an error/warning and a stack trace was sent, append to log.
+    if (['error', 'warn'].includes(level) && typeof localLogObject.stack_trace !== 'undefined') {
+      metadata.stack_trace = localLogObject.stack_trace;
+    }
+
+    // Check if we received a request object
+    if (localLogObject.request) {
+      // Include relevant chunks of the node.js request object. If we include
+      // whole object wholesale, the recursion creates gigantic, useless logs.
+      metadata.request = {
+        path: localLogObject.request.path || {},
+        query: localLogObject.request.query || {},
+        payload: localLogObject.request.payload || {},
+        headers: localLogObject.request.headers || {},
+        auth: localLogObject.request.auth || {},
+      };
+
+      // Try to automatically detect user unless logObject.user.id already exists.
+      if (logObject.request.params && logObject.request.params.currentUser) {
+        if (typeof metadata.user.id === 'undefined') {
+          metadata.user.id = logObject.request.params.currentUser._id.toString();
+        }
+      }
+
+      // Sanitize passwords
+      if (metadata.request.payload && metadata.request.payload.password) {
+        delete metadata.request.payload.password;
+      }
+      if (metadata.request.payload && metadata.request.payload.confirm_password) {
+        delete metadata.request.payload.confirm_password;
+      }
+      if (metadata.request.payload && metadata.request.payload.old_password) {
+        delete metadata.request.payload.old_password;
+      }
+      if (metadata.request.payload && metadata.request.payload.new_password) {
+        delete metadata.request.payload.new_password;
+      }
+
+      // Sanitize OAuth client secrets
+      if (metadata.request.query && typeof metadata.request.query.client_secret !== 'undefined') {
```

Comment:
I would propose to replace `typeof metadata.request.query.client_secret !== 'undefined'` and the equivalent line 77 and 94 by `typeof metadata.request.query.client_secret === 'string'`.

Reply:
nice. Yeah I guess there's no other primitive that can be received so it's more explicit to check before acting on the data as a string.